docs/specification/2025-03-26/basic/authorization.md

```diff
@@ -4,242 +4,391 @@ type: docs
-  in [RFC8414](https://datatracker.ietf.org/doc/html/rfc8414).
-- MCP server _SHOULD_ follow the OAuth 2.0 Authorization Server Metadata protocol.
-- MCP servers that do not support the OAuth 2.0 Authorization Server Metadata protocol,
-  _MUST_ support fallback URLs.
+- MCP clients **MUST** parse the `WWW-Authenticate` response to discover the issuer URL, and construct 
+  the metadata endpoint from that issuer identifier.  
+- MCP servers **MUST** use an established identity provider and include its issuer in 
+  *HTTP 401 Unauthorized* responses.  
+- MCP clients **SHOULD** evaluate or enable the user to evaluate the trustworthiness of identity 
+  providers.  
+- MCP servers with partial support for other identity providers or that implement their own 
+  authentication and authorization logic **SHOULD** follow the OAuth 2.0 Authorization Server Metadata 
+  protocol.  
+- MCP servers **MUST** respond with a `WWW-Authenticate` header with `issuer`, `scope`, and `resource` 
+  identifiers.
 
-The discovery flow is illustrated below:
+The discovery flow is illustrated below:  
 
 ```mermaid
 sequenceDiagram
-    participant C as Client
-    participant S as Server
-
-    C->>S: GET /.well-known/oauth-authorization-server
-    alt Discovery Success
-        S->>C: 200 OK + Metadata Document
-        Note over C: Use endpoints from metadata
-    else Discovery Failed
-        S->>C: 404 Not Found
-        Note over C: Fall back to default endpoints
-    end
-    Note over C: Continue with authorization flow
+    participant MC as MCP Client
+    participant MS as MCP Server
+    participant IDP as Identity Provider
+
+    MC->>MS: Request data
+    MS-->>MC: 401 Unauthorized with WWW-Authenticate
+    MC->>MC: Construct metadata endpoint from issuer identifier
+
+    Note over MC: Continue with authorization flow
 ```
 
 #### 2.3.1 Server Metadata Discovery Headers
 
-MCP clients _SHOULD_ include the header `MCP-Protocol-Version: <protocol-version>` during
-Server Metadata Discovery to allow the MCP server to respond based on the MCP protocol
-version.
+MCP clients **SHOULD** include the header `MCP-Protocol-Version: <protocol-version>` during Server 
+Metadata Discovery to allow the MCP server to respond based on the MCP protocol version.
 
 For example: `MCP-Protocol-Version: 2024-11-05`
 
-#### 2.3.2 Authorization Base URL
+### 2.4 Identity provider integration
+
+#### 2.4.1 Overview
+
+MCP servers **MAY** override endpoints on an as-needed basis if the identity provider requires 
+additional gestures that are not captured by the specification.
+
+MCP servers **MUST NOT** complete PKCE or other public client flows on behalf of the client.
+
+MCP servers that do not implement a [metadata discovery process](#23-server-metadata-discovery) must 
+be assumed to implement the OAuth flow through standard endpoints (e.g., `/authorize` and `/token`)
+
+#### 2.4.2 Flow Description
```

Comment:
This looks the same to me as the flow described in section 2.2. Am I missing something obvious, or is the only difference the URL of the IdP?

Reply:
You're right - I removed the duplication.